mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/mail/PostgresAttachmentModule.java

```diff
@@ -35,7 +35,7 @@ public interface PostgresAttachmentModule {
     interface PostgresAttachmentTable {
 
         Table<Record> TABLE_NAME = DSL.table("attachment");
-        Field<UUID> ID = DSL.field("id", SQLDataType.UUID.notNull());
+        Field<String> ID = DSL.field("id", SQLDataType.VARCHAR.notNull());
```

Comment:
:-(

Reply:
https://github.com/apache/james-project/pull/2048#discussion_r1502017082
https://github.com/apache/james-project/pull/2048#discussion_r1502032559

Reply:
How about being clever and reworking `AttachmentId`?

We could have:
 - An interface with a Factory for `AttachmentId`
 - Current AttachmentId class be renamed as` LegacyAttachmentId` and used by Cassandra app, JPA, memory
 - Have a new `UuidBackedAttachmentId` used in the Postgres app ?

Reply:
Move this issue to another ticket https://github.com/orgs/linagora/projects/2/views/2?pane=issue&itemId=56534403
This PR stay for too long and should be closed soon